platform/golden/cfg/preseed-debian-11.cfg

```diff
@@ -50,17 +51,36 @@ tasksel tasksel/skip-tasks  string standard
 tasksel tasksel/first multiselect openssh-server
 popularity-contest popularity-contest/participate boolean false
 
-d-i pkgsel/include string curl wget openssh-server neovim sudo cloud-init
+d-i pkgsel/include string curl wget openssh-server neovim sudo cloud-init ovmf grub-efi-amd64
 d-i pkgsel/language-pack-patterns   string
 d-i pkgsel/install-language-support boolean false
 d-i pkgsel/upgrade select full-upgrade
 
+# -------- Hardware Firmware --------
+d-i hw-detect/load_firmware boolean true
+
 ###########
 ## Use XFS
 
-d-i partman/default_filesystem string xfs
+# Force UEFI booting ('BIOS compatibility' will be lost). Default: false.
+d-i partman-efi/non_efi_system boolean true
+
+# Ensure the partition table is GPT - this is required for EFI
+d-i partman-partitioning/choose_label select gpt
+d-i partman-partitioning/default_label string gpt
+#d-i partman/default_filesystem string xfs
+
+# Use raw partitions 
 d-i partman-auto/method string regular
 
+# # Use LVM
+# d-i partman-auto/method string lvm
+# d-i partman-auto-lvm/guided_size string max
+# d-i partman-lvm/device_remove_lvm boolean true
+# d-i partman-md/device_remove_md boolean true
+# d-i partman-lvm/confirm boolean true
+# d-i partman-lvm/confirm_nooverwrite boolean true
```

Comment:
I understand why you made these changes. I spent 30+ hours trying to find an xfs configuration that works. I suspect you have spent a lot of time searching for this config. It looks like the real problem to be solved is

"How do we make a `gpt` partition layout with xfs?"

Reply:
This is probably more a remnant of the initial understanding that you had to manually create a partition map with a 1M free partition, a 1G FAT32 partition (or rather a FAT32 labeled partition), and the rest of the space as you saw fit.  But it turns out that if you get the EFI firmware loaded into the VM first, then everything else "works normally" from there.  That doesn't mean that I have an XFS file system for /, but at least the system seems to try to "do the right thing" with building the UEFI/grub config at the moment.